pkg/server/server.go

```diff
@@ -329,7 +356,15 @@ func (s *Server) Check(ctx context.Context, req *openfgapb.CheckRequest) (*openf
 		graph.WithMaxConcurrentReads(s.maxConcurrentReadsForCheck),
 	)
 
-	resp, err := checkResolver.ResolveCheck(ctx, &graph.ResolveCheckRequest{
+	cachedCheckResolver := graph.NewCachedCheckResolver(
+		checkResolver,
+		graph.WithExistingCache(s.checkCache),
+		graph.WithCacheTTL(s.checkQueryCacheTTL),
+	)
+
+	checkResolver.SetDelegate(cachedCheckResolver)
+
+	resp, err := cachedCheckResolver.ResolveCheck(ctx, &graph.ResolveCheckRequest{
```

Comment:
This should be behind an experimental flag

Reply:
Instead of experimental flag, I would argue that this should be behind configuration (i.e., allow user to choose whether to enable or not)

Reply:
It should be both enabled with an experimental flag AND configurable. The reason we put it behind the experimental flag is to express that it is not yet ready for production adoption. When we move it out of experimental we are communicating that we're confident in using it in production scenarios.

Reply:
Added experimental flag